### PR TITLE
node: devicemgr: remove obsolete pre-1.20 checkpoint file support

### PR DESCRIPTION
**Pull request by ffromani**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
In commit 2f426fdba6fa0458c4a47dd22b37cc920d9f2a98 we added compatibility (and tests) to deal with pre-1.20 checkpoint files. We can now remove this code, including the brittle (but still the best we had available for the issue at hand) e2e tests and simplify again the device manager code

#### Which issue(s) this PR fixes:
Fixes #N/A

#### Special notes for your reviewer:
see PR https://github.com/kubernetes/kubernetes/pull/102882 for details about the original issue

#### Does this PR introduce a user-facing change?
```release-note
the kubelet is no longer able to recover from device manager state file older than 1.20. If the proper recommended upgrade flow is followed, there should be no issue.
```

### CODE REVIEW

**Comment by ffromani** [comment]
@kannon92 FYI

**Comment by ffromani** [comment]
failed because (unrelated)
```
ERROR: (gcloud.compute.instances.create) Could not fetch resource:
 - Quota 'NVIDIA_K80_GPUS' exceeded.  Limit: 0.0 in region us-west1.
	metric name = compute.googleapis.com/nvidia_k80_gpus
	limit name = NVIDIA-K80-GPUS-per-project-region
	limit = 0.0
	dimensions = region: us-west1 
```

I think this is non-blocking

**Comment by ffromani** [comment]
/test pull-kubernetes-node-kubelet-serial-containerd


I still don't get how this change can break this test

**Comment by ffromani** [comment]
/test pull-kubernetes-node-kubelet-serial-containerd

still looks like infra issue to me

File: pkg/kubelet/cm/devicemanager/manager.go [hunk 1]
```diff
@@ -472,29 +472,19 @@ func (m *ManagerImpl) readCheckpoint() error {
 	// the current version first. Trying to restore older format checkpoints is
 	// relevant only in the kubelet upgrade flow, which happens once in a
 	// (long) while.
-	cp, err := m.getCheckpointV2()
+	cp, err := m.getCheckpoint()
```
Comment (swatisehgal) [inline pkg/kubelet/cm/devicemanager/manager.go:475]: We need to remove the comment above.

File: test/e2e_node/device_manager_test.go [hunk 2]
```diff
@@ -57,227 +49,13 @@ import (
...
-	checkpointName  = "kubelet_internal_checkpoint"
 )
 
 // Serial because the test updates kubelet configuration.
 var _ = SIGDescribe("Device Manager", framework.WithSerial(), feature.DeviceManager, nodefeature.DeviceManager, func() {
-	checkpointFullPath := filepath.Join(devicePluginDir, checkpointName)
 	f := framework.NewDefaultFramework("devicemanager-test")
 	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
 
-	ginkgo.Context("With SRIOV devices in the system", func() {
```
Comment (swatisehgal) [inline test/e2e_node/device_manager_test.go:69]: I think we should preserve an end to end test that tests out checkpoint restoration logic at least covering the positive test case where kubelet recovers the checkpoint state correctly.

File: pkg/kubelet/cm/devicemanager/manager.go (hunk 1 above)
Comment (ffromani) [reply pkg/kubelet/cm/devicemanager/manager.go:475]: fight, let me fix this

File: test/e2e_node/device_manager_test.go (hunk 2 above)
Comment (ffromani) [reply test/e2e_node/device_manager_test.go:69]: We have some form of coverage in the `device_plugin_test.go` e2e tests. If we need something more specific, I'd go for a separate PR. WDYT?

Comment (swatisehgal) [reply test/e2e_node/device_manager_test.go:69]: I think you might be right here, we have coverage in `device_plugin_test.go` so let's move on. Thanks!

**Review by swatisehgal — APPROVED** [review]
/lgtm

/assign @klueska
for approval

**Comment by SergeyKanzhelev** [comment]
> Does this PR introduce a user-facing change?

Yes it does. Since we are breaking this compatibility we should mention it in relnotes.

I don't think it is critical though. I cannot imagine people upgrading from 1.20 to 1.31 and expecting scheduled pods to keep running as normal - they will need to drain the node.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Comment by ffromani** [comment]
/hold

> > Does this PR introduce a user-facing change?
> 
> Yes it does. Since we are breaking this compatibility we should mention it in relnotes.
> 
> I don't think it is critical though. I cannot imagine people upgrading from 1.20 to 1.31 and expecting scheduled pods to keep running as normal - they will need to drain the node.

I see your point but we are breaking compatibility against direct upgrade from a version long unsupported. IOW, users upgrading using the suggested path should never see this issue.
That said I'm fine fixing the release notes.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Review by mrunalp — APPROVED** [review]

**Pull request merged**